src/main/java/bio/terra/service/StudyService.java

```diff
@@ -0,0 +1,89 @@
+package bio.terra.service;
+
+import bio.terra.controller.exception.ApiException;
+import bio.terra.dao.StudyDao;
+import bio.terra.exceptions.ValidationException;
+import bio.terra.flight.study.create.StudyCreateFlight;
+import bio.terra.flight.study.delete.StudyDeleteFlight;
+import bio.terra.model.DatasetSummaryModel;
+import bio.terra.model.StudyJsonConversion;
+import bio.terra.model.StudyModel;
+import bio.terra.model.StudyRequestModel;
+import bio.terra.model.StudySummaryModel;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.FlightState;
+import bio.terra.stairway.FlightStatus;
+import bio.terra.stairway.Stairway;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Component
+public class StudyService {
+    private final StudyDao studyDao;
+    private final Stairway stairway;
+    private final DatasetService datasetService;
+
+    @Autowired
+    public StudyService(StudyDao studyDao, Stairway stairway, DatasetService datasetService) {
+        this.studyDao = studyDao;
+        this.stairway = stairway;
+        this.datasetService = datasetService;
+    }
+
+    public StudySummaryModel createStudy(StudyRequestModel studyRequest) {
+        FlightMap flightMap = new FlightMap();
+        flightMap.put(JobMapKeys.REQUEST.getKeyName(), studyRequest);
+        flightMap.put(JobMapKeys.DESCRIPTION.getKeyName(), "Creating a study");
+        String flightId = stairway.submit(StudyCreateFlight.class, flightMap);
+        return getResponse(flightId, StudySummaryModel.class);
+    }
+
+    public StudyModel retrieve(UUID id) {
+        return StudyJsonConversion.studyModelFromStudy(studyDao.retrieve(id));
+    }
+
+    public List<StudySummaryModel> enumerate(int offset, int limit) {
+        return studyDao.enumerate(offset, limit)
+            .stream()
+            .map(summary -> StudyJsonConversion.studySummaryModelFromStudySummary(summary))
+            .collect(Collectors.toList());
+    }
+
+    public boolean delete(UUID id) {
+        List<DatasetSummaryModel> referencedDatasets = datasetService.getDatasetsReferencingStudy(id);
+        if (referencedDatasets == null || referencedDatasets.isEmpty()) {
+            FlightMap flightMap = new FlightMap();
+            flightMap.put(JobMapKeys.REQUEST.getKeyName(), id);
+            flightMap.put(JobMapKeys.DESCRIPTION.getKeyName(), "Deleting a study");
```

Comment:
nit: maybe put the study name in the description?

Reply:
I don't have the name at that point. Do you want me to retrieve the study summary to log the name?

Reply:
Oh, no not at all. Maybe the id? My thought was just recording something so what is in the stairway history is more meaningful.